src/features/profile/Profile.tsx

```diff
@@ -0,0 +1,94 @@
+import { useCallback, useEffect } from "react";
+import { Link, useParams } from "react-router-dom";
+import { useSelector } from "react-redux";
+import useProfileState from "./use-profile-state";
+import { AuthState } from "features/auth/store/types";
+import { fetchOneProfile } from "./services/profile.service";
+import ShowSkills from "./components/show-skills";
+import ShowExperiences from "./components/show-experiences";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import IconButton from "@mui/material/IconButton";
+import EditIcon from "@mui/icons-material/Edit";
+
+const Profile = () => {
+  const { profile, setProfile } = useProfileState();
+
+  const id = parseInt(useParams<RouteParams>().id!);
```

Comment:
¿Que pasa si no es un numero?, ¿que pasa si meten texto?, ¿que pasa si es un defined?

Quiza todo eso lo puedas contemplar con mejor detalle y dando mensajes de error si haces algo como
```suggestion
  const id = useParamId();
```
O algo quiza mas concreto...
```suggestion
  const id: number | null = useParamId();
  if (!id) return <ErrorComponent />; /* no estoy seguro de que esto se pueda hacer en esta parte del codigo, quiza debe ser ANTES del otro render por temas de hooks condicionales */
```

Algo mas concreto y quiza mas seguro seria:
```suggestion
  const id: number | null = useParamId();
  useEffect(() => {
     if (!id) redirect('/..../error');
  }, [id]);
```

Ahora, yo realmente pondria eso DENTRO de useParamId(); asi que volvemos al original xd
```suggestion
  const id = useParamId();
```

Reply:
pero wait, useParamId vendria a ser un componente extra? una funcion extra?

Reply:
Una funcion extra xddd, osea un custom-hook, estas obsesionado con componentes extras (?